### PR TITLE
Protect class instance invokable methods

### LINKED ISSUE #28630: Improve JSInvokable documentation in case of trimming.
There is no caution about JS Interop (Call .NET from JS) when **trimming** / **AOT** is enabled.

Linker has no idea about JS Interop and when there is no reference in project to the method with `JSInvokable`, this method will be removed in case of trimming / AOT and you will get a runtime exception about the missing method when JS tries to call this .NET method.

The docs should warn you about this and offer a solution.
For example, if you inspect aspnetcore [source code](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Web/src/Forms/InputFile/InputFileJsCallbacksRelay.cs), you can see how they deal with this by using `DynamicDependency` attribute.

```CSharp
internal sealed class InputFileJsCallbacksRelay : IDisposable
{
    private readonly IInputFileJsCallbacks _callbacks;

    public IDisposable DotNetReference { get; }

    [DynamicDependency(nameof(NotifyChange))]
    public InputFileJsCallbacksRelay(IInputFileJsCallbacks callbacks)
    {
        _callbacks = callbacks;

        DotNetReference = DotNetObjectReference.Create(this);
    }

    [JSInvokable]
    public Task NotifyChange(BrowserFile[] files)
        => _callbacks.NotifyChange(files);

    public void Dispose()
    {
        DotNetReference.Dispose();
    }
}
```
This is also important, because when you contribute to blazor open source projects the project owner might ask for documentation about this behavior.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 2964d558-cd6d-7421-f7b9-cb939792c2cf
* Version Independent ID: a3517e56-69fc-1ec7-b55a-dbb54d68a807
* Content: [Call .NET methods from JavaScript functions in ASP.NET Core Blazor](https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-dotnet-from-javascript?view=aspnetcore-7.0)
* Content Source: [aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #28630

Thanks @ScarletKuro! 🚀 ... This might be a bit *rough*, but I couldn't wait until next Tuesday. See what you think of this. After we get the lingo hammered out sufficiently, I'll ping for engineering review.

You'll see some repeats of text here. It's just due to a type of versioning that we use. Ignore the duplicate sections/text.

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/components/class-libraries.md](https://github.com/dotnet/AspNetCore.Docs/blob/0e11d66277d5a6cd6a9bb46fdc11e32df7351a12/aspnetcore/blazor/components/class-libraries.md) | [aspnetcore/blazor/components/class-libraries](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/class-libraries?branch=pr-en-us-28631) |
| [aspnetcore/blazor/host-and-deploy/webassembly.md](https://github.com/dotnet/AspNetCore.Docs/blob/0e11d66277d5a6cd6a9bb46fdc11e32df7351a12/aspnetcore/blazor/host-and-deploy/webassembly.md) | [aspnetcore/blazor/host-and-deploy/webassembly](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?branch=pr-en-us-28631) |
| [aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md](https://github.com/dotnet/AspNetCore.Docs/blob/0e11d66277d5a6cd6a9bb46fdc11e32df7351a12/aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md) | [aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-dotnet-from-javascript?branch=pr-en-us-28631) |

### CODE REVIEW

**Comment by ScarletKuro** [comment]
Looks great and clear

**Comment by mkArtakMSFT** [comment]
Thanks @guardrex.
@MackinnonBuck can you please review this? Thanks!

File: aspnetcore/blazor/components/class-libraries.md [hunk 1]
```diff
@@ -353,6 +353,10 @@ Blazor enables JavaScript isolation in standard [JavaScript modules](https://dev
 
 For more information, see <xref:blazor/js-interop/call-javascript-from-dotnet#javascript-isolation-in-javascript-modules>.
 
+## Avoid trimming JavaScript-invokable .NET methods
+
+[Runtime relinking](xref:blazor/host-and-deploy/webassembly#runtime-relinking) trims class instance JavaScript-invokable .NET methods unless they're protected. For more information, see <xref:blazor/js-interop/call-dotnet-from-javascript#avoid-trimming-javascript-invokable-net-methods>.
```
Comment (MackinnonBuck) [inline aspnetcore/blazor/components/class-libraries.md:358]: Nit: Maybe this is a stretch, but I think the term "protected" might be misinterpreted as having to do with the `protected` access modifier. Perhaps we could say something like this?

```suggestion
[Runtime relinking](xref:blazor/host-and-deploy/webassembly#runtime-relinking) trims class instance JavaScript-invokable .NET methods unless they're explicitly preserved. For more information, see <xref:blazor/js-interop/call-dotnet-from-javascript#avoid-trimming-javascript-invokable-net-methods>.
```

**Review by MackinnonBuck — APPROVED** [review]
Just a small nit, but this looks great!

**Pull request merged**